internal/config/profile_integration_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	atlasProfile   = "atlas"
+	newProfileName = "newProfileName"
+)
+
+func createProfile(profileContents string) error {
+	p.fs = afero.NewMemMapFs()
+
+	testConfigDir, err := configHome()
+	if err != nil {
+		return err
+	}
+
+	filename := fmt.Sprintf("%v/%v.toml", testConfigDir, ToolName)
+
+	err = afero.WriteFile(p.fs, filename, []byte(profileContents), 0600)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createProfileWithOneNonDefaultUser() error {
+	const contents = `
+[atlas]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneDefaultUserOneNonDefault() error {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+
+[atlas]
+  base_url = "http://atlas.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithFullDescription() error {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_url = "http://om_url.com/"
+  ops_manager_ca_certificate = "/path/to/certificate"
+  ops_manager_skip_verify = "false"
+  public_api_key = "some_public_key"
+  private_api_key = "some_private_key"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+  service = "cloud"
+`
+	return createProfile(contents)
+}
+
+func TestProfile_List(t *testing.T) {
+	err := createProfileWithOneDefaultUserOneNonDefault()
+	if err != nil {
+		t.Error(err)
```

Comment:
```suggestion
		t.Fatal(err)
```
If you fail to create the profile then the rest of the test are going to fail as well